### DISCUSSION TITLE
"Millions of connections" on a VPS

### DISCUSSION THREAD

**Discussion by RamonZoocha** [uNetworking/uWebSockets.js#477] [Q&A] [answered]
Hey guys, sorry if my question is dumb, but I'm not DevOps or anything.

I've read this article by Alex Hultman https://alexhultman.medium.com/millions-of-active-websockets-with-node-js-7dc575746a01.

It says he had to set up a range of IPs on his local WebSockets server, then connect 20k clients to each IP.

My question is:

- is it possible to setup multiple public IP addresses on a remote VPS server (not aws, just a bare linux server), or do I depend on this feature being provided by my VPS provider?

It would also be nice to hear from ppl who have already used this method on remote servers. Did it go well?

Thanks!

**Comment by ghost** [comment, marked as answer]
Hi, TCP uses a 16-bit port number. So there is a limit of 65k connections **per identical IP pair**.

If you accept connections **from wildly different hosts on the internet** your server only needs 1 IP

**Reply by RamonZoocha** [reply]
That's awesome, thanks for your work!

**Reply by johnnyApplePRNG** [reply]
Define "wildly"?

Isn't just a single address away enough?

18.19.20.21 + 18.19.20.22 = 130k connections?
